jonason/Parser.hpp

```diff
@@ -0,0 +1,45 @@
+//
+//  Parser.hpp
+//  jonason
+//
+//  Created by Gabriel Lumbi on 2023-03-03.
+//
+
+#ifndef Parser_hpp
+#define Parser_hpp
+
+#include <string>
+#include <memory>
+
+#include "JSONValue.hpp"
+#include "Tokenizer.hpp"
+
+struct JSONValue;
+
+namespace jonason {
+
+struct ParseError: public std::exception {
+    enum Code {
+        UNEXPECTED_EOF = 1,
+        UNEXPECTED_TOKEN = 2,
+        UNKNOWN = 999
+    };
+
+    Code code;
+    std::string message;
+
+    explicit ParseError(Code code, const std::string& message) : code(code), message(message) {};
+
+    explicit operator bool() { return code; }
```

Comment:
Errors are `true`?

Reply:
Yeah, so I can do `if (error) { /* handle error */ }`

Reply:
Isn't the fact that you have a ParseError instance enough to know that you have an error?

Reply:
I had something before were it always returned this `ParseError` object, it was more like a `ParseResult`, anyway, irrelevant now so I should just remove this `operator bool()`